src/test/java/org/broadinstitute/hellbender/utils/pileup/PileupElementUnitTest.java

```diff
@@ -179,13 +258,171 @@ public void testCreatePileupForReadAndOffset1(){
+                {read, 0, true, false, null, D},
+                {read, 1, false, false, null, D},
+                {read, 2, false, false, null, D},
+                {read, 3, false, false, null, D},
+                {read, 4, false, false, null, D},
+                {read, 5, false, false, null, D},
+                {read, 6, false, false, null, D},
+                {read, 7, false, false, null, D},
+                {read, 8, false, false, null, D},
+                {read, 9, false, true, null, D},
+                {read, 10, true, false, P, X},
+                {read, 11, false, false, P, X},
+                {read, 12, false, false, P, X},
+                {read, 13, false, false, P, X},
+                {read, 14, false, false, P, X},
+                {read, 15, false, false, P, X},
+                {read, 16, false, false, P, X},
+                {read, 17, false, false, P, X},
+                {read, 18, false, false, P, X},
+                {read, 19, false, true, P, X},
+                {read, 20, true, false, EQ, M},
+                {read, 21, false, false, EQ, M},
+                {read, 22, false, false, EQ, M},
+                {read, 23, false, false, EQ, M},
+                {read, 24, false, false, EQ, M},
+                {read, 25, false, false, EQ, M},
+                {read, 26, false, false, EQ, M},
+                {read, 27, false, false, EQ, M},
+                {read, 28, false, false, EQ, M},
+                {read, 29, false, true, EQ, M},
+                {read, 30, true, false, X, S},
+                {read, 31, false, false, X, S},
+                {read, 32, false, false, X, S},
+                {read, 33, false, false, X, S},
+                {read, 34, false, false, X, S},
+                {read, 35, false, false, X, S},
+                {read, 36, false, false, X, S},
+                {read, 37, false, false, X, S},
+                {read, 38, false, false, X, S},
+                {read, 39, false, true, X, S},
+        };
+    }
+
+    @Test(dataProvider = "elementData")
+    public void testOperators(GATKRead read, int goodOffset, boolean atStartOfCurrentCigar, boolean atEndOfCurrentCigar, CigarOperator prev, CigarOperator next) throws Exception {
+        final PileupElement pe = PileupElement.createPileupForReadAndOffset(read, goodOffset);
+        Assert.assertEquals(pe.atStartOfCurrentCigar(), atStartOfCurrentCigar, "atStartOfCurrentCigar");
+        Assert.assertEquals(pe.atEndOfCurrentCigar(), atEndOfCurrentCigar, "atEndOfCurrentCigar");
+        Assert.assertEquals(pe.getAdjacentOperator(PileupElement.Direction.PREV), prev, "prev");
+        Assert.assertEquals(pe.getAdjacentOperator(PileupElement.Direction.NEXT), next, "next");
+        for (final CigarOperator op : CigarOperator.values()) {
+            Assert.assertEquals(pe.isImmediatelyAfter(op), atStartOfCurrentCigar && op == prev);
+            Assert.assertEquals(pe.isImmediatelyBefore(op), atEndOfCurrentCigar && op == next);
+        }
```

Comment:
No need for a loop here -- replace with two ifs for clarity:

    if ( atStartOfCurrentCigar ) {
        Assert.assertTrue(pe.isImmediatelyAfter(prev);
    }

    if ( atEndOfCurrentCigar ) {
        Assert.assertTrue(pe.isImmediatelyBefore(next);
    }